core/node/state_keeper/src/tests/mod.rs

```diff
@@ -500,3 +500,31 @@ async fn protocol_upgrade() {
         .run(sealer)
         .await;
 }
+
+/// Checks the next L2 block timestamp is updated upon receiving a transaction
+#[tokio::test]
+async fn l2_block_timestamp_updated_after_first_tx() {
+    let config = StateKeeperConfig {
+        transaction_slots: 2,
+        ..StateKeeperConfig::default()
+    };
+    let sealer = SequencerSealer::with_sealers(config, vec![Box::new(SlotsCriterion)]);
+    let fake_timestamp = 555;
+
+    TestScenario::new()
+        .seal_l2_block_when(|updates| updates.l2_block.executed_transactions.len() == 1)
+        .next_tx("First tx", random_tx(1), successful_exec())
+        .l2_block_sealed("L2 block 1")
+        .update_l2_block_timestamp("Update the next l2 block timestamp", fake_timestamp)
+        .next_tx("New tx", random_tx(1), successful_exec())
+        .l2_block_sealed_with("L2 block 2", move |updates| {
+            let actual = updates.l2_block.timestamp;
+            assert_eq!(
+                actual, fake_timestamp,
+                "L2 block timestamp. Expected {}, got {}",
+                fake_timestamp, actual
```

Comment:
Nit: `fake_timestamp` and `actual` may be inlined in the string above. But this looks redundant in any case because `assert_eq!` prints the compared values on failure.

Reply:
fixed